src/wallet/mod.rs

```diff
@@ -158,63 +140,78 @@ impl fmt::Display for AddressInfo {
     }
 }
 
-#[derive(Debug, Default)]
-/// Options to a [`sync`].
-///
-/// [`sync`]: Wallet::sync
-pub struct SyncOptions {
-    /// The progress tracker which may be informed when progress is made.
-    pub progress: Option<Box<dyn Progress>>,
+#[derive(Debug, Clone, PartialEq)]
+/// Error applying an update to the wallet.
+pub enum ApplyUpdateErr {
+    /// The data in the update was somehow incompatible with the existing chain's data.
+    Chain(UpdateFailure),
+    /// The update was compatible but it was missing **full** transactions for some txids it found.
```

Comment:
Just to make sure that I understand: this error is thrown if the Update is containing a certain `txidA` in the `new_txids`, but there's not a transaction with txid equal to `txidA` in the blocks between the last valid checkpoint and `new_tip`. Is that right?

Edit: I tried to read `apply_update` and concluded that I'm probably wrong

Reply:
It is thrown when we are told there is a relevant `txid` in some block but we haven't provided the actual transaction. This comes up with electrum which first returns you a list of `txids` which you would go and turn into an update. The wallet would then tell you which transactions it actually needs you to download with that error. You go and fetch them and re-apply it.